sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs

```diff
@@ -37,37 +37,100 @@ public RepositoryHandler(Uri repositoryUri, ClientDiagnostics clientdiagnostics,
             return await ProcessAsync(new List<string>() { dtmi }, cancellationToken).ConfigureAwait(false);
         }
 
-        public async Task<IDictionary<string, string>> ProcessAsync(IEnumerable<string> dtmis, CancellationToken cancellationToken)
+        public IDictionary<string, string> Process(string dtmi, CancellationToken cancellationToken)
+        {
+            return Process(new List<string>() { dtmi }, cancellationToken);
+        }
+
+        public IDictionary<string, string> Process(IEnumerable<string> dtmis, CancellationToken cancellationToken)
```

Comment:
As of yet, this method is the exact replica of the async method with the exception of calling Fetch instead of FetchAsync
I am speaking to a few people to see what is the right way to do this, duplicating this much code just to change a sync vs async call is really expensive. 

for now, I will leave the PR in this state so other things can be reviewed while I investigate different options.

Reply:
using bool async pattern 
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs#L45
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs#L55-L61

Reply:
Nice. And good find of prior art.

Reply:
This seems like a reasonable way to do it. :)